applications/FluidDynamicsApplication/custom_utilities/mass_conservation_utility.cpp

```diff
@@ -30,131 +29,211 @@ namespace Kratos
 
     double pos_vol = 0.0;
     double neg_vol = 0.0;
     double inter_area = 0.0;
-    const auto& r_comm = mrModelPart.GetCommunicator().GetDataCommunicator();
+    mrModelPart.GetCommunicator().SynchronizeVariable(DISTANCE);
+    mrModelPart.GetCommunicator().SynchronizeVariable(VELOCITY);
+    mrModelPart.GetCommunicator().SynchronizeVariable(MESH_VELOCITY);
 
     ComputeVolumesAndInterface( pos_vol, neg_vol, inter_area );
 
-    this->mInitialPositiveVolume = r_comm.SumAll(pos_vol);
-    this->mInitialNegativeVolume = r_comm.SumAll(neg_vol);
+    this->mInitialPositiveVolume = pos_vol;
+    this->mInitialNegativeVolume = neg_vol;
     this->mTheoreticalNegativeVolume = neg_vol;
-    inter_area = r_comm.SumAll(inter_area);
 
     std::string output_line =   "------ Initial values ----------------- \n";
     output_line +=              "  positive volume (air)   = " + std::to_string(this->mInitialPositiveVolume) + "\n";
     output_line +=              "  negative volume (water) = " + std::to_string(this->mInitialNegativeVolume) + "\n";
     output_line +=              "  fluid interface (area)  = " + std::to_string(inter_area) + "\n\n";
     output_line +=              "------ Time step values --------------- \n";
     output_line +=              "sim_time \t\twater_vol \t\tair_vol \t\twater_err \t\tnet_inf \t\t";
-    output_line +=              "net_outf \t\tint_area \t\tcorr_shift \n";
+    output_line +=              "net_outf \t\tint_area \n";
 
     return output_line;
 }
 
 
-std::string MassConservationCheckProcess::ExecuteInTimeStep(){
+
+std::string MassConservationUtility::ComputeBalancedVolume(){
 
     // performing all necessary calculations
     double pos_vol = 0.0;
     double neg_vol = 0.0;
-    double inter_area = 0;
+    double inter_area = 0.0;
+    mrModelPart.GetCommunicator().SynchronizeVariable(DISTANCE);
+    mrModelPart.GetCommunicator().SynchronizeVariable(VELOCITY);
+    mrModelPart.GetCommunicator().SynchronizeVariable(MESH_VELOCITY);
+
     ComputeVolumesAndInterface( pos_vol, neg_vol, inter_area );
     double net_inflow_inlet = ComputeFlowOverBoundary(INLET);
     double net_inflow_outlet = ComputeFlowOverBoundary(OUTLET);
+    mInterfaceArea = inter_area;
 
-    // computing global quantities via MPI communication
-    const auto& r_comm = mrModelPart.GetCommunicator().GetDataCommunicator();
-    std::vector<double> local_data{pos_vol, neg_vol, inter_area, net_inflow_inlet, net_inflow_outlet};
-    std::vector<double> remote_sum{0, 0, 0, 0, 0};
-    r_comm.SumAll(local_data, remote_sum);
-
-    pos_vol = remote_sum[0];
-    neg_vol = remote_sum[1];
-    inter_area = remote_sum[2];
-    net_inflow_inlet = remote_sum[3];
-    net_inflow_outlet = remote_sum[4];
-
-    // making a "time step forwards" and updating the
-    const double current_time = mrModelPart.GetProcessInfo()[TIME];
+    const double current_time = mrModelPart.GetProcessInfo()[*mpTimeVariable];
```

Comment:
```suggestion
    const auto& r_process_info = mrModelPart.GetProcessInfo();
    const double current_time = r_process_info[*mpTimeVariable];
```
(just in case...)